circom-prover/src/prover.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::Result;
+use num::BigUint;
+use std::thread::JoinHandle;
+
+pub mod arkworks;
+pub mod serialization;
+
+pub struct CircomProof {
+    pub proof: Vec<u8>,
+    pub pub_inputs: Vec<u8>,
+}
+
+pub enum ProofLib {
+    Arkworks,
+    RapidSnark,
+}
+
+pub fn prove(
+    lib: ProofLib,
+    zkey_path: String,
+    witnesses: JoinHandle<Vec<BigUint>>,
+) -> Result<CircomProof> {
+    match lib {
+        ProofLib::Arkworks => arkworks::generate_circom_proof(zkey_path, witnesses),
+        ProofLib::RapidSnark => panic!("Not supported yet."),
+    }
+}
```

Comment:
I think it can be controlled by feature flags
the dependencies will be built when users enable one feature/or two features

Reply:
the same as the witness calculator

Reply:
fixed in [265abb6](https://github.com/zkmopro/mopro/pull/304/commits/265abb65315544aa49003ca2e878f62fa9037f2c)